projects/notebooks/visualizations/rhods-notebooks-performance/analyze/__init__.py

```diff
@@ -1,17 +1,60 @@
 import json
 import logging
+import numpy as np
 
-import matrix_benchmarking.common as common
+from typing import List
 
+import matrix_benchmarking.common as common
+import matrix_benchmarking.regression as regression
+import matrix_benchmarking.regression.zscore as zscore
 
 def run():
-    logging.info(f"Received {common.Matrix.count_records()} new entries")
-    for entry in common.Matrix.all_records():
-        pass
 
     logging.info(f"Received {common.LTS_Matrix.count_records()} historic LTS entries")
-    for lts_entry in common.LTS_Matrix.all_records(): pass
+    lts_entries = []
+    # Take the first result if they were gathered
+    for entry in common.LTS_Matrix.all_records():
+        if entry.results and type(entry.results) is list:
+            entry.results = entry.results[0]
+            lts_entries.append(entry)
+        else:
+            lts_entries.append(entry)
```

Comment:
I would prefer something like this:
```suggestion        
        data = extract_data_from_gathered(entry) if entry.is_gathered() \
                      else extract_data_from_entry(entry)
```
in particular, this:
```
entry.results = entry.results[0]
```
is a bad practice [in Python]: you receive an object (from the LTS_Matrix), you should not modify it, or you don't know what the future will become 😆 
In Java, `entry.results` would be private, and the compiler would error out on this line, but nothing like that happens in Python!

example, if there were multiple Regression objects, like there are multiple Plotting classes. Your code executes happily, then mine executes and crashes, because `entry.is_gathered -> True` but `entry.results --> not a list` 🙃 

==> I think you should remove this for loop and just have this:
```
lts_entries = list(common.LTS_Matrix.all_records())
```
and include the handling of gathered entries in matbench `regression` package

Reply:
Gotcha! I'll try implementing `deepcopy` on the `MatrixEntry` then I'll provide the functionality to flatten it in the `regression` package. Let me know if you had some other method in mind :)

Reply:
no, don't `deepcopy` the entry, try not to modify it and use it as it is
unless you find any fundamental flaw, but the `gathered` aspect is an important concept from my POV